lib/AstToMiniRust.ml

```diff
@@ -589,6 +589,19 @@ and translate_array (env: env) is_toplevel (init: Ast.expr): env * MiniRust.expr
 (* However, generally, we will have a type provided by the context that may
    necessitate the insertion of conversions *)
 and translate_expr_with_type (env: env) (e: Ast.expr) (t_ret: MiniRust.typ): env * MiniRust.expr =
+  let rec erase_lifetime_info (t: MiniRust.typ) = match t with
```

Comment:
can you do this with a visitor? it might be more future-proof

Reply:
Fixed in [0007e91](https://github.com/FStarLang/karamel/pull/441/commits/0007e91fed9d8c7b131e432e3a636c4638ffc3c7). As discussed in person, this required a few tweaks to the MiniRust AST to generate visitors for typ.